app/Models/Beatmapset.php

```diff
@@ -562,6 +562,22 @@ public function setApproved($state, $user)
             ->update(['approved' => $this->approved]);
     }
 
+    public function disqualify($user)
+    {
+        if (!$this->isQualified()) {
+            return false;
+        }
+
+        DB::transaction(function () use ($user) {
+            $this->setApproved('pending', $user);
+            $this->userRatings()->delete();
```

Comment:
Better in `setApproved` method itself so it can be used on both disqualify and qualify.